package/src/views/SkiaView.web.tsx

```diff
@@ -0,0 +1,167 @@
+import React from "react";
+import type { LayoutChangeEvent, ViewProps } from "react-native";
+import { View } from "react-native";
+import type { Canvas, Surface } from "canvaskit-wasm";
+
+import type { SkRect } from "../skia";
+import type { SkiaValue } from "../values";
+
+import type { DrawingInfo, DrawMode, RNSkiaDrawCallback } from "./types";
+
+export interface SkiaViewProps extends ViewProps {
+  /**
+   * Sets the drawing mode for the skia view. There are two drawing
+   * modes, "continuous" and "default", where the continuous mode will
+   * continuously redraw the view, and the default mode will only
+   * redraw when any of the regular react properties are changed like
+   * sizes and margins.
+   */
+  mode?: DrawMode;
+  /**
+   * When set to true the view will display information about the
+   * average time it takes to render.
+   */
+  debug?: boolean;
+  /**
+   * Draw callback. Will be called whenever the view is invalidated and
+   * needs to redraw. This is either caused by a change in a react
+   * property, a touch event, or a call to redraw. If the view is in
+   * continuous mode the callback will be called 60 frames per second
+   * by the native view.
+   */
+  onDraw?: RNSkiaDrawCallback;
+}
+
+let NativeIdCounter = 1000;
+
+export class SkiaView extends React.Component<
+  SkiaViewProps,
+  { width: number; height: number }
+> {
+  constructor(props: SkiaViewProps) {
+    super(props);
+    this._nativeId = NativeIdCounter++;
+    this._surface = null;
+    this.state = { width: -1, height: -1 };
+  }
+
+  private _nativeId: number;
+  private _surface: Surface | null;
+  private _unsubscriptions: Array<() => void> = [];
+  //private _mode: DrawMode = "default";
+
+  private getKey() {
+    return `rnskia-${this._nativeId}`;
+  }
+
+  private unsubscribeAll() {
+    this._unsubscriptions.forEach((u) => u());
+    this._unsubscriptions = [];
+  }
+
+  private onLayout(evt: LayoutChangeEvent) {
+    this.setState({
+      width: evt.nativeEvent.layout.width,
+      height: evt.nativeEvent.layout.height,
+    });
+    this.redraw();
+  }
+
+  componentDidMount() {
+    this.redraw();
+  }
+
+  componentWillUnmount() {
+    this.unsubscribeAll();
+  }
+
+  componentDidUpdate() {
+    if (this.state.width > -1) {
+      // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+      // @ts-ignore
```

Comment:
Should CanvasKit be a prop here? With the default prop being global.CanvasKit and throw if both are undefined